pkg/plugin/plugin.go

```diff
@@ -69,57 +69,74 @@ type ConfigOption struct {
 	Max types.Int `json:"max,omitempty"`
 }
 
-// Info contains plugin information.
+// Info contains channel plugin information.
 type Info struct {
-	Type             string          `db:"type" json:"-"`
-	Name             string          `db:"name" json:"name"`
-	Version          string          `db:"version" json:"version"`
-	Author           string          `db:"author" json:"author"`
-	ConfigAttributes json.RawMessage `db:"config_attrs" json:"config_attrs"` // ConfigOption(s) as json-encoded list
+	// Type of the channel plugin.
+	//
+	// Not part of the JSON object. Will be set to the channel plugin file name before database insertion.
+	Type string `db:"type" json:"-"`
+	// Name of this channel plugin in a human-readable value.
+	Name string `db:"name" json:"name"`
+	// Version of this channel plugin.
+	Version string `db:"version" json:"version"`
+	// Author of this channel plugin.
+	Author string `db:"author" json:"author"`
+	// ConfigAttributes contains multiple ConfigOption(s) as JSON-encoded list.
+	ConfigAttributes json.RawMessage `db:"config_attrs" json:"config_attrs"`
 }
 
 // TableName implements the contracts.TableNamer interface.
 func (i *Info) TableName() string {
 	return "available_channel_type"
 }
 
+// Contact to receive notifications for the NotificationRequest.
 type Contact struct {
 	FullName  string     `json:"full_name"`
 	Addresses []*Address `json:"addresses"`
 }
 
+// Address to receive this notification. Each Contact might have multiple addresses.
 type Address struct {
 	Type    string `json:"type"`
 	Address string `json:"address"`
 }
 
+// Object which this NotificationRequest is all about, e.g., a service check.
```

Comment:
I don't think a `service check` is an object.

Reply:
The text says that the object might be _about_ a service check, not being a service check.

Reply:
> The text says that the object might be about a service check

That sentence doesn't make sense to me either. An `Event` could be about a service check, but I don't quite understand what an object being about a service check is supposed to mean.

Reply:
Then please take a look at the example `object`: <https://github.com/Icinga/icinga-notifications/blob/17cb89f00b9618c118bc91e2689dd98c65567980/doc/10-Channels.md#example-sendnotification-request>.

I am also open for alternative suggestions.

Reply:
Or was your point that it does not refer to a service check but more to a service object?

_Btw, I find the name `object` hard to grasp, especially as both in Icinga 2 and OOP lingo everything is a object._